runtime/src/qio/qbuffer.c

```diff
@@ -148,14 +148,8 @@ qioerr _qbytes_init_iobuf(qbytes_t* ret)
 {
   void* data = NULL;
   
-  // allocate 4K-aligned (or page size aligned)
-  // multiple of 4K
-  data = qio_valloc(qbytes_iobuf_size);
+  data = qio_memalign(sys_page_size(), qbytes_iobuf_size);
```

Comment:
Looks good. For 4K pages it's worth page aligning these (so certain system calls might be faster) but it's probably better to save ~2MB of space than to 2MB-align them. It's also less likely that these are communicated. If later on it turns out that it matters that these are really page aligned, we could conceivably put them on a different heap from the main one if hugepages are in use, so their page size can really be 4K.  (e.g. using the system allocator). But we don't have to do that now.